instrumented/intrumented.go

```diff
@@ -0,0 +1,142 @@
+package instrumented
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/uw-labs/substrate"
+)
+
+var labels = []string{"status", "topic"}
+
+// AsyncMessageSink is an instrumented message sink
+// The counter vector will have the labels "status" and "topic"
+type AsyncMessageSink struct {
+	impl       substrate.AsyncMessageSink
+	counter    *prometheus.CounterVec
+	topic      string
+	bufferSize int
+}
+
+// NewAsyncMessageSink returns a pointer to a new AsyncMessageSink
+func NewAsyncMessageSink(sink substrate.AsyncMessageSink, counterOpts prometheus.CounterOpts, topic string, bufferSize int) *AsyncMessageSink {
+	counter := prometheus.NewCounterVec(counterOpts, labels)
+
+	if err := prometheus.Register(counter); err != nil {
+		if are, ok := err.(prometheus.AlreadyRegisteredError); ok {
+			counter = are.ExistingCollector.(*prometheus.CounterVec)
+		} else {
+			panic(err)
+		}
+	}
+
+	return &AsyncMessageSink{
+		impl:       sink,
+		counter:    counter,
+		topic:      topic,
+		bufferSize: bufferSize,
+	}
+}
+
+// PublishMessages implements message publshing wrapped in instrumentation
+func (ams *AsyncMessageSink) PublishMessages(ctx context.Context, acks chan<- substrate.Message, messages <-chan substrate.Message) (rerr error) {
+	successes := make(chan substrate.Message, ams.bufferSize)
```

Comment:
How about you do 
```
successes := make(chan substrate.Message, cap(acks))
```
instead so that you don't have to pass in buffer size because it will confuse people?

Reply:
Same for the source